packages/manager/apps/pci-load-balancer/src/api/hook/useGateways.ts

```diff
@@ -20,3 +29,63 @@ export const useGetSubnetGateways = (
     enabled: !!projectId && !!regionName && !!subnetId,
     throwOnError: true,
   });
+
+export const useRegionPlans = (regionName: string) => {
+  const { data: project } = useProject();
+  const { data } = useProductAvailability(project.project_id, {
+    addonFamily: 'gateway',
+  });
+
+  return useMemo(
+    () =>
+      data?.plans
+        .filter(({ code }) => code.includes('hour'))
+        .filter((plan) =>
+          plan.regions.some((region) => region.name === regionName),
+        )
+        .map((item) => item.code) || [],
+    [data, regionName],
+  );
+};
+
+const getGatewaySize = (addon: TAddon) => addon.product.split('-').pop();
+
+type TGatewayCatalog = {
+  size: string;
+  price: number;
+};
+
+const getSmallestGateway = (
+  catalog: TCatalog,
+  plans: string[],
+): TGatewayCatalog => {
+  const sizeOrder = ['s', 'm', 'l', 'xl', '2xl'];
+
+  const gateways = catalog.addons
+    .filter((addon) => plans.includes(addon.planCode))
+    .slice()
+    .sort((a, b) => {
+      const sizeA = getGatewaySize(a);
+      const sizeB = getGatewaySize(b);
+
+      return sizeOrder.indexOf(sizeA) - sizeOrder.indexOf(sizeB);
+    });
+
+  return {
+    size: getGatewaySize(gateways[0]),
+    price: gateways[0].pricings[0].price,
+  };
+};
+
+// TODO: similar use case in floating IP, private network and compute, think to move it to pci-common
+export const useSmallestGatewayByRegion = (region: string) => {
+  const { ovhSubsidiary } = useContext(ShellContext).environment.getUser();
+  const plans = useRegionPlans(region);
+
+  return useQuery({
+    queryKey: ['gateway', 'catalog', region, ovhSubsidiary],
+    queryFn: () => getCatalog(ovhSubsidiary),
+    select: (catalog) => getSmallestGateway(catalog, plans),
+    enabled: plans.length > 0 && !!region,
```

Comment:
With this you make the queries impossible to be fetch in parallel and you will have a big loading